rust/src/hlil/function.rs

```diff
@@ -0,0 +1,100 @@
+use core::hash::{Hash, Hasher};
+
+use binaryninjacore_sys::BNFreeHighLevelILFunction;
+use binaryninjacore_sys::BNGetHighLevelILBasicBlockList;
+use binaryninjacore_sys::BNGetHighLevelILInstructionCount;
+use binaryninjacore_sys::BNGetHighLevelILOwnerFunction;
+use binaryninjacore_sys::BNGetHighLevelILSSAForm;
+use binaryninjacore_sys::BNHighLevelILFunction;
+use binaryninjacore_sys::BNNewHighLevelILFunctionReference;
+
+use crate::basicblock::BasicBlock;
+use crate::function::Function;
+use crate::rc::{Array, Ref, RefCountable};
+
+use super::{HighLevelILBlock, HighLevelILInstruction};
+
+pub struct HighLevelILFunction {
+    pub(crate) handle: *mut BNHighLevelILFunction,
+}
+
+unsafe impl Send for HighLevelILFunction {}
+unsafe impl Sync for HighLevelILFunction {}
+
+impl Eq for HighLevelILFunction {}
+impl PartialEq for HighLevelILFunction {
+    fn eq(&self, rhs: &Self) -> bool {
+        self.handle == rhs.handle
+    }
+}
+
+impl Hash for HighLevelILFunction {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.handle.hash(state);
+    }
+}
+
+impl HighLevelILFunction {
+    pub(crate) unsafe fn from_raw(handle: *mut BNHighLevelILFunction) -> Self {
```

Comment:
It might be better to change this to `ref_from_raw` so we don't accidentally forget to wrap this in a `Ref<>` other places in the API.

Reply:
Like the hash, this is also legacy from the llil. I'll remove it.

Reply:
Should I also change that in LLIL and MLIL? Maybe create a new PR?

Reply:
If you fix it in HLIL, I don't mind copying it over to MLIL/LLIL. If you want to do it, you're welcome to. In this PR or in another doesn't make a difference to me.

Reply:
It's mostly just a timebomb/footgun. Doesn't need to block merging this PR if you don't change it.

Reply:
I solved that by making all `from_raw` functions return `Ref<Self>`.

While working on it I realized that binja `Ref` is kind of an unnecessary abstraction, because all types that use `Ref` are always owned and only need to exist and a `Ref<Self>`. In this case you might just make the type own it's handle by implementing `Clone` and `Drop` directly, instead of implement `RefCountable` and make it a `Ref<Self>`.

But I diverge, and that's beyond the scope for this PR.